src/components/AddTodoForm/AddTodoForm.tsx

```
@@ -0,0 +1,109 @@
+import { useState } from 'react';
+import { Nullable } from '../..//components/domain/Nullable';
+import usersFromServer from '../../api/users';
+import { Todo } from '../domain/Todo';
+
+type AddTodoFormProps = {
+  onSubmit: (todo: Omit<Todo, 'id'>) => void;
+};
+
+export const AddTodoForm = ({ onSubmit }: AddTodoFormProps) => {
+  const [title, setTitle] = useState('');
+  const [titleError, setTitleError] = useState<Nullable<string>>(null);
+
+  const [userId, setUserId] = useState(0);
+  const [userIdError, setUserIdError] = useState<Nullable<string>>(null);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newTitle = event.target.value;
+
+    setTitle(newTitle);
+
+    if (titleError && newTitle.trim().length > 0) {
+      setTitleError(null);
+    }
+  };
+
+  const handleUserIdChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const newUserId = +event.target.value;
+
+    setUserId(newUserId);
+
+    if (userIdError && newUserId !== 0) {
```

Similar issue for the user select change handler: the code only clears the user error when `newUserId !== 0`: `if (userIdError && newUserId !== 0) { setUserIdError(null); }`. The checklist says: "Hide an error message immediately after any change in the corresponding field." Clear the error immediately on any change (when `userIdError` exists) instead of requiring a non-default value.